Modules/Loadable/VesselEditor/Logic/vtkSlicerVesselEditorLogic.cxx

```diff
@@ -405,4 +405,36 @@ ::SplitTubeInSpatialObject(vtkMRMLSpatialObjectsNode* spatialNode, int tubeID, d
       break;
     }
   }
-}
\ No newline at end of file
+}
+
+//---------------------------------------------------------------------------
+void vtkSlicerVesselEditorLogic
+::RenumberTubesInSpatialObject(vtkMRMLSpatialObjectsNode* spatialNode)
+{
+  if (!spatialNode)
+  {
+    return ;
+  }
+
+  TubeNetType* spatialObject = spatialNode->GetSpatialObject();
+  char childName[] = "Tube";
+  TubeNetType::ChildrenListType* tubeList =
+    spatialObject->GetChildren(spatialObject->GetMaximumDepth(), childName);
```

Comment:
How about creating a function in TubeMath that does the renumbering?   This
seems to be mixing logic and GUI elements of the program inappropriately.

On Wed, Mar 30, 2016 at 10:39 AM, sumedhasingla <notifications@github.com>
wrote:

> In Modules/Loadable/VesselEditor/Logic/vtkSlicerVesselEditorLogic.cxx
> <https://github.com/KitwareMedical/VesselView/pull/161#discussion_r57899785>
> :
>
> > \ No newline at end of file
> > +}
> > +
> > +//---------------------------------------------------------------------------
> > +void vtkSlicerVesselEditorLogic
> > +::RenumberTubesInSpatialObject(vtkMRMLSpatialObjectsNode* spatialNode)
> > +{
> > +  if (!spatialNode)
> > +  {
> > +    return ;
> > +  }
> > +
> > +  TubeNetType* spatialObject = spatialNode->GetSpatialObject();
> > +  char childName[] = "Tube";
> > +  TubeNetType::ChildrenListType* tubeList =
> > +    spatialObject->GetChildren(spatialObject->GetMaximumDepth(), childName);
>
> I am using it before the tree creation. So am not considering parent child
> relationship and just assigning all the tubes to one single parent.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/KitwareMedical/VesselView/pull/161/files/d7e98353fc6971a924397696de266ff32110d0e7#r57899785>
>



-- 
====================================================
Stephen R. Aylward, Ph.D.
Senior Director of Operations, North Carolina, Kitware, Inc.
http://www.kitware.com and http://www.aylward.org
(919) 969-6990 x300
====================================================